osdocs15191: ContentX - Improving "Understanding networking" chapter (#96874)
openshift/openshift-docs: modules/nw-understanding-networking-core-layers-and-components.adoc

// Module included in the following assemblies:
//
// * networking/understanding-networking.adoc

:_mod-docs-content-type: CONCEPT
[id="nw-understanding-networking-core-layers-and-components_{context}"]
= Core network layers and components

> Networking in {product-title} is built on two fundamental layers: the `pod network` and the `service network`. The pod network is where your applications live. The service network makes your applications reliably accessible.

Reviewer: ```suggestion
{openshift-networking} is built on two fundamental layers: the `pod network` and the `service network`. The pod network is where your applications live. The service network makes your applications reliably accessible.
```

> == The pod network

Reviewer 2: All titles and section headings must have an anchor ID.

OCP docs guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-titles-and-section-headings

Author: Oh geez, sorry about that.

> The pod network is a flat network space where every pod in the cluster receives its own unique IP address. This network is managed by the Container Network Interface (CNI) plugin. The CNI is responsible for wiring each pod into the cluster network.

Reviewer 2: ```suggestion
The pod network is a flat network space where every pod in the cluster receives its own unique IP address. This network is managed by the Container Network Interface (CNI) plugin. The CNI plugin is responsible for wiring each pod into the cluster network.
```

> This design allows pods to communicate directly with each other using their IP addresses, regardless of which node they are running on. However, these pod IPs are ephemeral. This means the IPs are destroyed when the pod is destroyed and a new IP is assigned when a new pod is created. Because of this, you should never rely on pod IP addresses directly for long-lived communication.

Reviewer: ```suggestion
This design allows pods to communicate directly with each other using their IP addresses, regardless of which node they are running on. However, these pod IPs are ephemeral. This means the IPs are destroyed when the pod is destroyed and a new IP address is assigned when a new pod is created. Because of this, you should never rely on pod IP addresses directly for long-lived communication.
```

> == The service network

Reviewer 2: All titles and section headings must have an anchor ID.

OCP docs guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-titles-and-section-headings

A service is a networking object that provides a single, stable virtual IP address, called a ClusterIP, and a DNS name for a logical group of pods.

> When a request is sent to a service's ClusterIP, {product-title} automatically load-balances the traffic to one of the healthy pods backing that service. It uses labels and selectors to keep track of which pods belong to which service. This abstraction makes your applications resilient because individual pods can be created or destroyed without affecting the applications trying to reach them.

Reviewer: ```suggestion
When a request is sent to a service's ClusterIP, {product-title} automatically load-balances the traffic to one of the healthy pods backing that service. It uses Kubernetes labels and selectors to keep track of which pods belong to which service. This abstraction makes your applications resilient because individual pods can be created or destroyed without affecting the applications trying to reach them.
```
